src/main/java/org/mastodon/mamut/feature/spot/ellipsoid/SpotEllipsoidFeature.java

```diff
@@ -0,0 +1,190 @@
+import org.mastodon.feature.Dimension;
+import org.mastodon.feature.Feature;
+import org.mastodon.feature.FeatureProjection;
+import org.mastodon.feature.FeatureProjectionKey;
+import org.mastodon.feature.FeatureProjectionSpec;
+import org.mastodon.feature.FeatureProjections;
+import org.mastodon.feature.FeatureSpec;
+import org.mastodon.feature.Multiplicity;
+import org.mastodon.feature.io.FeatureSerializer;
+import org.mastodon.io.FileIdToObjectMap;
+import org.mastodon.io.ObjectToFileIdMap;
+import org.mastodon.io.properties.DoublePropertyMapSerializer;
+import org.mastodon.mamut.model.Spot;
+import org.mastodon.properties.DoublePropertyMap;
+import org.scijava.plugin.Plugin;
+
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.util.LinkedHashMap;
+import java.util.LinkedHashSet;
+import java.util.Map;
+import java.util.Set;
+
+import static org.mastodon.feature.FeatureProjectionKey.key;
+
+/**
+ * Feature that computes the 3 semi-axes and the volume from the covariance matrix of {@link Spot}s.
+ * <p>
+ * @author Stefan Hahmann
+ */
+public class SpotEllipsoidFeature implements Feature<Spot>, FeatureSerializer<SpotEllipsoidFeature, Spot> {
+
+	public static final String KEY = "Spot ellipsoid properties";
+
+	private static final String HELP_STRING =
+			"Computes spot ellipsoid parameters, i.e. the 3 semi axes and the volume. ";
+
+	private final Map<FeatureProjectionKey, FeatureProjection<Spot>> projectionMap;
+
+	public static final FeatureProjectionSpec SHORT_SEMI_AXIS_PROJECTION_SPEC =
+			new FeatureProjectionSpec("Short semi-axis", Dimension.LENGTH);
+
+	public static final FeatureProjectionSpec MIDDLE_SEMI_AXIS_PROJECTION_SPEC =
+			new FeatureProjectionSpec("Middle semi-axis", Dimension.LENGTH);
+
+	public static final FeatureProjectionSpec LONG_SEMI_AXIS_PROJECTION_SPEC =
+			new FeatureProjectionSpec("Long semi-axis", Dimension.LENGTH);
+
+	public static final FeatureProjectionSpec VOLUME_PROJECTION_SPEC =
+			new FeatureProjectionSpec("Volume", Dimension.NONE);
+
+	final DoublePropertyMap<Spot> shortSemiAxis;
+
+	final DoublePropertyMap<Spot> middleSemiAxis;
+
+	final DoublePropertyMap<Spot> longSemiAxis;
+
+	final DoublePropertyMap< Spot > volume;
+
+	@Plugin( type = FeatureSpec.class )
+	public static class SpotEllipsoidFeatureSpec extends FeatureSpec< SpotEllipsoidFeature, Spot >
+	{
+		public SpotEllipsoidFeatureSpec()
+		{
+			super(KEY, HELP_STRING, SpotEllipsoidFeature.class, Spot.class, Multiplicity.SINGLE,
+					SHORT_SEMI_AXIS_PROJECTION_SPEC, MIDDLE_SEMI_AXIS_PROJECTION_SPEC, LONG_SEMI_AXIS_PROJECTION_SPEC, VOLUME_PROJECTION_SPEC);
+		}
+	}
+
+	public static final SpotEllipsoidFeatureSpec SPOT_ELLIPSOID_FEATURE_SPEC = new SpotEllipsoidFeatureSpec();
+
+	SpotEllipsoidFeature(
+			final DoublePropertyMap<Spot> shortSemiAxis, final DoublePropertyMap<Spot> middleSemiAxis,
+			final DoublePropertyMap<Spot> longSemiAxis, final DoublePropertyMap<Spot> volume) {
```

Comment:
Please put the curly bracket on a new line. Or is this a problem with the mastodon coding style?

Reply:
I was formatting using the mastodon coding style. I formatted once more now with this style, but startet the parameter section already on the line, where the constructor starts.